plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/CopyPropertiesAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.unomi.plugins.baseplugin.actions;
+
+import org.apache.commons.beanutils.BeanUtilsBean;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.unomi.api.Event;
+import org.apache.unomi.api.PropertyType;
+import org.apache.unomi.api.actions.Action;
+import org.apache.unomi.api.actions.ActionExecutor;
+import org.apache.unomi.api.services.EventService;
+import org.apache.unomi.api.services.ProfileService;
+import org.apache.unomi.persistence.spi.PropertyHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CopyPropertiesAction implements ActionExecutor {
+
+    private static final Logger logger = LoggerFactory.getLogger(CopyPropertiesAction.class);
+    private ProfileService profileService;
+
+    public void setProfileService(ProfileService profileService) {
+        this.profileService = profileService;
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public int execute(Action action, Event event) {
+        boolean changed = false;
+
+        for (Map.Entry<String, Object> entry : getEventPropsToCopy(action, event).entrySet()) {
+            // propType Check
+            PropertyType propertyType = profileService.getPropertyType(entry.getKey());
+            String propertyName = "properties." + entry.getKey();
+            if (propertyType != null && propertyType.isMultivalued()) {
+                changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValue");
+            } else {
+                Object profileProperty = event.getProfile().getProperty(entry.getKey());
+                if (profileProperty == null) {
+                    if (entry.getValue() instanceof List) {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValues");
+                    } else {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "alwaysSet");
+                    }
```

Comment:
This Should depends on the propType, in case the propType exist but is not multivalued you have an issue here:

Input:
- event input prop: toto = ["newValue"]
- propType: singleValue
- profile.properties.toto = null

Output:
-profile.properties.toto = ["newValue"]

Should be:
- FAIL: cannot copy a multiValue from event to a propType that is not multi valued